src/pip/_internal/req/req_install.py

```diff
@@ -927,14 +928,11 @@ def install(
         install_options = list(install_options) + \
             self.options.get('install_options', [])
 
-        if self.isolated:
-            # https://github.com/python/mypy/issues/1174
-            global_options = global_options + ["--no-user-cfg"]  # type: ignore
-
         with TempDirectory(kind="record") as temp_dir:
             record_filename = os.path.join(temp_dir.path, 'install-record.txt')
             install_args = self.get_install_args(
-                global_options, record_filename, root, prefix, pycompile,
+                global_options, self.isolated, record_filename, root, prefix,
```

Comment:
Do you need to be passing `self.isolated` to `get_install_args()`, or can you just access `self.isolated` from within that method?